getting-started/setup-building.rst

```diff
@@ -58,6 +58,24 @@ itself. Git is easily available for all common operating systems.
   ``git push``, or ``git fetch``.  On Windows, you should also
   :ref:`enable autocrlf <autocrlf>`.
 
+.. _install-pre-commit:
+
+Install pre-commit
+------------------
+
+To make sure your code is formatted correctly, we recommend installing
+`pre-commit <https://pre-commit.com>`_ and setting it up as a Git hook:
+
+1. Go to https://pre-commit.com#installation.
+
+2. Run the install command for your global package manager.
+
+3. Install the Git hook::
+
+      $ pre-commit install
```

Comment:
If you install this, then commit on a branch that doesn't have the pre-commit config file (3.10 and earlier), you get an error:


```console
$ git commit -m TEMP
No .pre-commit-config.yaml file was found
- To temporarily silence this, run `PRE_COMMIT_ALLOW_NO_CONFIG=1 git ...`
- To permanently silence this, install pre-commit with the --allow-missing-config option
- To uninstall pre-commit run `pre-commit uninstall`
```

Is it worth suggesting this instead? Committing on a pre-3.11 branch is unlikely but possible.

```suggestion
      $ pre-commit install --allow-missing-config
```